core-web/libs/block-editor/src/lib/extensions/ai-image-prompt/ai-image-prompt.store.ts

```diff
@@ -5,38 +5,46 @@ import { Injectable } from '@angular/core';
 
 import { switchMap, tap, withLatestFrom } from 'rxjs/operators';
 
-import { ComponentStatus, DotCMSContentlet } from '@dotcms/dotcms-models';
+import { ComponentStatus } from '@dotcms/dotcms-models';
 
 import { PromptType } from './ai-image-prompt.models';
 
-import { DotAiService } from '../../shared';
+import { AIImageSize, DotAiService } from '../../shared';
+import { AIImagePrompt, DotGeneratedAIImage } from '../../shared/services/dot-ai/dot-ai.models';
 
 const DEFAULT_INPUT_PROMPT: PromptType = 'input';
 
 export interface DotAiImagePromptComponentState {
     showDialog: boolean;
-    selectedPromptType: PromptType | null;
     prompt: string | null; // we always have the final prompt here
     editorContent: string | null;
-    contentlets: DotCMSContentlet[] | [];
+    images: DotGeneratedAIImage[];
     status: ComponentStatus;
     error: string;
+    selectedImage: DotGeneratedAIImage | null;
```

Comment:
The image showed to the user is not the selected? Why we need galleryActiveIndex?

Reply:
The `selectedImage` is the one the user pick to be added to the editor, the `galleryActiveIndex` is the image that is being display in the popup at the moment, they user can change it before select the final one.